### PR TITLE
[docs] Add an example for mocking cross-domain responses

### LINKED ISSUE #3016: [docs] Add 'How to mock cross-domain request using a RequestMock' example
Customer have problems during setup a `RequestMock` for cross-domain requests:
* https://github.com/DevExpress/testcafe/issues/1134#issuecomment-431857172
* https://github.com/DevExpress/testcafe/issues/2614
* https://github.com/DevExpress/testcafe/issues/1134

Need to create a separate example for it.

### PR DESCRIPTION
**Pull request by VasilyStrelyaev**
\cc @miherlosev 

closes #3016

### CODE REVIEW

File: docs/articles/documentation/test-api/intercepting-http-requests/mocking-http-responses.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ Mocking is useful when the tested app uses infrastructure that is difficult to d
 * [The onRequestTo Method](#the-onrequestto-method)
 * [The respond Method](#the-respond-method)
   * [A Custom Response Function](#a-custom-response-function)
+* [Examples](#examples)
+  * [Mocking Cross-Domain Responses](#mocking-cross-domain-responses)
```
Comment (miherlosev) [inline docs/articles/documentation/test-api/intercepting-http-requests/mocking-http-responses.md]: `[Mocking Cross-Domain Requests](#mocking-cross-domain-requests)`

File: docs/articles/documentation/test-api/intercepting-http-requests/mocking-http-responses.md [hunk 2]
```diff
@@ -117,3 +119,18 @@ Property | Type | Description
 Method | Description
 ------ | ---------------
 `setBody(value)` | Sets the response body.  Accepts a string as a parameter.
+
+## Examples
+
+### Mocking Cross-Domain Responses
+
+When you mock a cross-domain response (a response to a request to a different domain), set the response's `access-control-allow-origin` header to the tested site's domain (or an asterisk `*`) to pass [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) validation.
```
Comment (miherlosev) [inline docs/articles/documentation/test-api/intercepting-http-requests/mocking-http-responses.md]: `When you mock a cross-domain requests, ...`.

Cross-domain request is a common term. You don't need explain this.

There are two most popular cases:
* if 'access-control-allow-credentials' is specified, then you should  setup an exact domain for 'access-control-allow-origin' header. With `*` specified, the CORS will fail in this case.
* if 'access-control-allow-credentials' is not specified, then you can specify the exact domain or `*`.

File: docs/articles/documentation/test-api/intercepting-http-requests/mocking-http-responses.md (hunk 1 above)
Comment (VasilyStrelyaev) [reply docs/articles/documentation/test-api/intercepting-http-requests/mocking-http-responses.md]: well, the topic is called `Mocking HTTP Responses` so it says we're mocking responses not requests.

Comment (VasilyStrelyaev) [reply docs/articles/documentation/test-api/intercepting-http-requests/mocking-http-responses.md]: Is the topic title wrong?

Comment (miherlosev) [reply docs/articles/documentation/test-api/intercepting-http-requests/mocking-http-responses.md]: As I know, for this case the phrase `Mocking http requests` is more used than `Mocking http responses'.
See module descriptions in https://www.npmjs.com/search?q=mocking%20requests.

**Comment by VasilyStrelyaev** [comment]
Renamed the topic, reflected changes regarding 'mock requests not responses' and added info about `access-control-allow-credentials`.

Hope everything is correct now.

**Pull request merged**